configfs/fakertmr/fakertmr.go

```diff
@@ -140,15 +148,27 @@ func (r *RtmrSubsystem) MkdirTemp(dir, pattern string) (string, error) {
 	if p.Entry != "" {
 		return "", fmt.Errorf("MkdirTemp: rtmr entry %q cannot have subdirectories", dir)
 	}
-
-	if r.Entries == nil {
-		r.Entries = make(map[string]*rtmrEntry)
+	if err = os.MkdirAll(r.Path, 0755); err != nil {
+		return "", fmt.Errorf("MkdirTemp: %v", err)
 	}
 	name := configfsi.TempName(r.Random, pattern)
-	if _, ok := r.Entries[name]; ok {
-		return "", os.ErrExist
+	fakeRtmrPath := path.Join(r.Path, name)
+	if err = os.Mkdir(fakeRtmrPath, 0755); err != nil {
+		return "", fmt.Errorf("MkdirTemp: %v", err)
 	}
-	r.Entries[name] = &rtmrEntry{initialized: false, rtmrMaps: r.RtmrMaps}
+	// Create empty index, digest and tcg_map files.
+	if err = createEmptyFile(filepath.Join(fakeRtmrPath, tsmPathIndex)); err != nil {
+		return "", fmt.Errorf("MkdirTemp: %v", err)
+	}
+
+	if err = createEmptyFile(filepath.Join(fakeRtmrPath, tsmRtmrDigest)); err != nil {
+		return "", fmt.Errorf("MkdirTemp: %v", err)
+	}
+
+	if err = createEmptyFile(filepath.Join(fakeRtmrPath, tsmPathTcgMap)); err != nil {
+		return "", fmt.Errorf("MkdirTemp: %v", err)
+	}
+
```

Comment:
Your permissions I believe are incorrect, and you can simplify and inline createEmptyFile.
Index and digest are read-write. Only tcg_map is read-only. I don't know why you used 0666 if you're specifying RDONLY. The TSM also has default 0 privileges to the files, so only root would get access. I think for the test you should be using owner-only privileges of 0600 or 0400 depending on rw or ro.
```suggestion
// Create empty index, digest and tcg_map files.
perms := []int{os.RDWR, os.RDWR, os.O_RDONLY}
modes := []os.FileMode{0600, 0600, 0400}
for i, attr := range []string{tsmPathIndex, tsmRtmrDigest, tsmPathTcgMap} {
	p := filepath.Join(fakeRtmrPath, attr)
	f, err := os.OpenFile(p, perms[i]|os.O_CREATE, modes[i]); err != nil {
		return "", fmt.Errorf("MkdirTemp %q: %v", p, err)
	}
	f.Close()
}
```

Reply:
Done. Thanks for the suggestion.